File: src/Maptable.java
import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Queue;

/**
 * Created by songchiyun on 2017. 4. 12..
 */
public class Maptable {


    private int count;
    HashMap<String, Integer> map = new HashMap<>();
    HashMap<Integer, String> ItoSmap = new HashMap<>();
    private int[][] table = new int[8][10000000];
    private int index;
    Maptable(){
        index = 0;
    }

    public HashMap<Integer, String> getItoSmap() {
        return ItoSmap;
    }

    public void setItoSmap(HashMap<Integer, String> itoSmap) {
        ItoSmap = itoSmap;
    }

    public void addCount(int category, String feature){

        int position = getIndex(feature);
        if(position == index){

            table[category][position] = 1;
            map.put(feature, position);
            ItoSmap.put(position,feature);
            index++;

        }
        else{
            table[category][position]++;

        }

    }
    private int getIndex(String feature){

        if(map.containsKey(feature)){
            return map.get(feature);
        }
        else{
            return index;
        }
    }

    public int[][] getTable(){
        return this.table;
    }
    public HashMap<String, Integer> getMap(){
        return map;
    }
    public int getIndex(){
        return index;
    }

    public List<String> getSplit(String text){

        String[] feature = text.split(" ");
        List<String> splitList = new ArrayList<>();

        for(int i = 0; i < feature.length; i++){

            if(!splitList.contains(feature[i])){
                splitList.add(feature[i]);
            }
        }

        return splitList;
    }

}

File: python/mapTable.py
from konlpy.tag import Kkma
class mapTable(object):

    def __init__(self):
        self.kkma = Kkma()
        self.table = [[0] * 8 for i in range(10000000)]
        self.index = 0
        self.map = {}
        self.Index2String = {}

    def getMap(self):
        return self.map

    def getTable(self):
        return self.table

    def addCount(self, category, feature):
        position = self._getIndex(feature)
        #print(category, ":",feature)

        if position == self.index:
            self.table[position][category] = 1
            self.map[feature] =  position
            self.Index2String[position] = feature
            self.index += 1
        else:
            self.table[position][category] += 1

    def getIndex(self):
        return self.index

    def _getIndex(self, feature):
        if feature in self.map.keys():
            return self.map[feature]
        else:
            return self.index

    #use konlp libray
    def split(self, text):
        kkma = Kkma()
        feature = kkma.nouns(text)
        return feature